app-cpp/app355.cpp

```diff
@@ -177,7 +177,7 @@ std::vector<std::vector<app355::OrbitalData>> SatOrbitSingle::OnCreateTrains(con
             if (deltaMotion < 0.001)
             {
                 trainVector[i].insert(trainVector[i].end(), trainVector[j].begin(), trainVector[j].end());
-                trainVector.erase(trainVector.begin() + j);
+                trainVector.erase(trainVector.begin() + static_cast<std::ptrdiff_t>(j));
```

Comment:
interesting....
you are starting with a `unsigned i, j ` because you're iterating over a container (correct).
but then you downsample to a `signed`.
Q: what happens if the value to too-big to be represented as a `signed`?

you should probably assert that:
```
    assert(static_cast<std::ptrdiff_t>(j) >= 0);
    trainVector.erase(trainVector.begin() + static_cast<std::ptrdiff_t>(j));
```